libs/content/workspace-ffi/src/apple/ios_ffi.rs

```diff
@@ -707,7 +706,20 @@ pub unsafe extern "C" fn selection_rects(
         cont_start = new_end.selection.end();
     }
 
-    UITextSelectionRects { size: selection_rects.len() as i32, rects: selection_rects.as_ptr() }
+    UITextSelectionRects {
+        size: selection_rects.len() as i32,
+        rects: Box::into_raw(selection_rects.into_boxed_slice()) as *const CRect,
```

Comment:
Why does this need to be heap allocated at all?

Reply:
aka why is it a `const *` instead of just included directly?

Reply:
`Vec` doesn't have a C representation, so if I tried to return a Vec, Swift wouldn't be able to handle it.

Also, when I heap allocate it (done within the `into_boxed_slice`), it will remove all the excess space the `Vec` may have. It means the `.length` corresponds to the size of the array.